.github/workflows/branch-protection.yml

```
@@ -1,124 +1,158 @@
-            fi
-          else
-            echo "Not targeting main branch"
+          # Check that package.json is valid
+          if ! npm ls --json >/dev/null 2>&1; then
+            echo "❌ package.json has dependency issues"
+            exit 1
           fi
 
-      - name: Check for long-lived branches
+      - name: Check for breaking changes
         run: |
```

actionlint (1.7.11)

[error] "github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)